cmd/system-shutdown/system-shutdown-utils.c

```diff
@@ -66,6 +66,27 @@ void die(const char *msg)
 	exit(1);
 }
 
+int sc_read_reboot_arg(char *arg, size_t max_size)
+{
+	FILE *f;
+
+	// This file is used by systemd to pass around a reboot parameter See
+	// https://github.com/systemd/systemd/blob/v229/src/basic/def.h#L44
+	f = fopen("/run/systemd/reboot-param", "r");
+	if (!f)
+		return -1;
+
+	if (!fgets(arg, max_size, f)) {
+		fclose(f);
+		return -1;
+	}
+	arg[strcspn(arg, "\n")] = '\0';
```

Comment:
strchr() would need a check for NULL pointer on failure to find '\n', which is not needed by strcspn()